src/Components/Modifications/Stickytoc.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * File containing the Sticky TOC class
+ *
+ * This file is part of the MediaWiki skin Chameleon.
+ *
+ * @copyright 2013 - 2023
+ * @license   GPL-3.0-or-later
+ *
+ * The Chameleon skin is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option) any
+ * later version.
+ *
+ * The Chameleon skin is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @file
+ * @ingroup Skins
+ */
+
+namespace Skins\Chameleon\Components\Modifications;
+
+/**
+ * Class Sticky TOC
+ *
+ * @since 1.0
+ * @ingroup Skins
+ */
+class Stickytoc extends Modification {
```

Comment:
I'm not sure that this should be a Modification. If you look at the other Modifications, they are generic and intended to be used for modifying other components, e.g. making something else sticky or hidden. Specifically, by placing the modification node inside the component node to be affected in the XML.

In this case Stickytoc is not really modifying the parent node (`<cell>` in `stickytoc.xml`). While it is technically making changes to what is inside MainContent, it does not work in the same way as other Modifications.